core/kt-osrd-sim-infra/src/main/kotlin/fr/sncf/osrd/sim_infra/api/LoadingGaugeConstraint.kt

```diff
@@ -0,0 +1,8 @@
+package fr.sncf.osrd.sim_infra.api
+
+import fr.sncf.osrd.railjson.schema.rollingstock.RJSLoadingGaugeType
+
+interface LoadingGaugeConstraint {
+    /** Returns true if a train of the given type is compatible  */
+    fun isCompatibleWith(trainType: RJSLoadingGaugeType): Boolean
```

Comment:
This would require a duplication of the enum, because the serializable one needs `@Json` annotation. This requires switches to convert it back and forth (with a dependency one way or another). This does not sound more maintainable.

We could move the gauge enum (as well as linestring) to the infra package, and just remove `RJS` from its name, while still using it for serialization. Would that be a good compromise?

Reply:
> We could move the gauge enum (as well as linestring) to the infra package, and just remove RJS from its name, while still using it for serialization. Would that be a good compromise?

it would be bad in another way, as now, you'd need the infra package to use railjson

Reply:
> his requires switches to convert it back and forth (with a dependency one way or another).

Not really: you can have package A (railjson), package B (infra), and package C (glue code), which uses both

Reply:
An other idea would be to make loading gauge types opaque using StaticIdx. this way, indices can be converted to and from enums regardless of the underlying type.

Reply:
And then, LoadingGaugeConstraints can just be a StaticIdxSet

Reply:
I've changed it into ids.

We're losing a lot of code readability and type safety with this change in the java side (and with value types in general). I really hope the full kt migration will come soon, this interoperability doesn't really feel maintainable